File: src/macros/stun_slam.py
from simulate.key_press import key_press
from simulate.mouse.left_click import left_click
from utils.precise_sleep import precise_sleep

def stun_slam(slots):
  axe_slot: str = slots["axe"]
  mace_slot: str = slots["mace"]

  key_press(axe_slot)
  left_click()
  precise_sleep(1/20)
  key_press(mace_slot)
  left_click()

File: src/macros/stunSlam.cpp
#include <Windows.h>
#include <string>
#include <vector>
#include "../core/core.h"
#include "macros.h"

/**
 * Stun Slamming is when you disable your opponent's shield mid air and hit them with a mace
 */
void stunSlam(std::vector<std::string> keys) {
  std::string axe = keys[0];
  std::string mace = keys[1];

  keyPress(axe);
  leftClick();

  Sleep(50);  // 1 tick

  keyPress(mace);
  leftClick();
}